### PR TITLE
Eitanya/ai azure openai

### PR DESCRIPTION
**Pull request by EItanya**
# Description

Add the API for token based auth with Azure OpenAI in the AI Gateway.

## API changes

- Added Azure OpenAI as an option for the AI Upstream
- Added Azure OpenAI as an option for the embedding API for Semantic Cache and RAG

# Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: projects/gloo/api/v1/enterprise/options/ai/ai.proto [hunk 1]
```diff
@@ -96,6 +96,21 @@ message UpstreamSpec {
     CustomHost custom_host = 2;
   }
 
+  // Settings for the Azure OpenAI API
+  message AzureOpenAI {
+    // Auth Token to use for the OpenAI API
```
Comment (shashankram) [inline projects/gloo/api/v1/enterprise/options/ai/ai.proto:101]: nit:
```suggestion
    // Auth Token to use for the Azure OpenAI API
```

File: projects/gloo/api/v1/enterprise/options/ai/ai.proto [hunk 2]
```diff
@@ -96,6 +96,21 @@ message UpstreamSpec {
...
+    // This token will be placed into the `api-key` header
+    oneof auth_token_source {
+      SingleAuthToken auth_token = 1;
```
Comment (shashankram) [inline projects/gloo/api/v1/enterprise/options/ai/ai.proto:104]: missing comment?

**Review by shashankram — APPROVED** [review]

File: projects/gloo/api/v1/enterprise/options/ai/ai.proto [hunk 3]
```diff
@@ -96,6 +96,23 @@ message UpstreamSpec {
...
 
+  // Settings for the Azure OpenAI API
+  message AzureOpenAI {
+    // Auth Token to use for the Azure OpenAI API
+    // This token will be placed into the `api-key` header
+    oneof auth_token_source {
+      // Auth Token to use for the OpenAI API
+      // This token will be placed into the `api-key` header
+      SingleAuthToken auth_token = 1;
+      // use AD or other workload identity mechanism
```
Comment (jenshu) [inline projects/gloo/api/v1/enterprise/options/ai/ai.proto:107]: what is this comment referring to?

File: projects/gloo/api/v1/enterprise/options/ai/ai.proto (hunk 2 above)
Comment (jenshu) [reply projects/gloo/api/v1/enterprise/options/ai/ai.proto:106]: should this comment be updated? it's nearly the same as the oneof comment

File: projects/gloo/api/v1/enterprise/options/ai/ai.proto (hunk 3 above)
Comment (EItanya) [reply projects/gloo/api/v1/enterprise/options/ai/ai.proto:107]: It's meant as a TODO, I can add that.

**Review by jenshu — APPROVED** [review]

**Pull request merged**